### ISSUE TITLE
oiiotool: set_channelnames method not updating output_channelformats

### ISSUE THREAD

**Issue by kazuki-takahashi** [OpenImageIO/oiio#1562]
### Problem

"read" methods creates a map information with bit depth and channel name, but set_channelnames method doesn't update the map after renaming channel names, and oiiotool converts channels to 32 bit since the new channel names are not contained in the output_channelformats.

For example, if I have the following single multi-layer exr file that is mix of half/float,
```
    diffuse.b, 16-bit floating-point, sampling 1 1
    diffuse.g, 16-bit floating-point, sampling 1 1
    diffuse.r, 16-bit floating-point, sampling 1 1
    specular.b, 32-bit floating-point, sampling 1 1
    specular.g, 32-bit floating-point, sampling 1 1
    specular.r, 32-bit floating-point, sampling 1 1
```

and rename channels by using --chnames like this,
```
oiiotool input.exr --ch "diffuse.r,diffuse.g,diffuse.b" --chnames "__diffuse.r,__diffuse.g,__diffuse.b" -o output.exr
```

the new name channels get converted to 32 bit.
```
    __diffuse.b, 32-bit floating-point, sampling 1 1
    __diffuse.g, 32-bit floating-point, sampling 1 1
    __diffuse.r, 32-bit floating-point, sampling 1 1
```

**Expected behavior:** [What you expected to happen]

Keeping the original bit depth

**Actual behavior:** [What actually happened]

Converting to 32 bit

### Steps to Reproduce

1. create an image having 16-bit/32-bit channels 
```
oiiotool --create 320x240 6 --chnames diffuse.r,diffuse.g,diffuse.b,specular.r,specular.g,specular.b -d diffuse.r=half -d diffuse.g=half -d diffuse.b=half -d specular.r=float -d specular.g=float -d specular.b=float -o ./input.exr
```

2. check bit depth
```
exrheader ./input.exr
    diffuse.b, 16-bit floating-point, sampling 1 1
    diffuse.g, 16-bit floating-point, sampling 1 1
    diffuse.r, 16-bit floating-point, sampling 1 1
    specular.b, 32-bit floating-point, sampling 1 1
    specular.g, 32-bit floating-point, sampling 1 1
    specular.r, 32-bit floating-point, sampling 1 1
```

3. rename by --chnames
```
oiiotool input.exr --ch "diffuse.r,diffuse.g,diffuse.b" --chnames "__diffuse.r,__diffuse.g,__diffuse.b" -o output.exr
```

4. check bit depth
```
exrheader ./output.exr
    __diffuse.b, 32-bit floating-point, sampling 1 1
    __diffuse.g, 32-bit floating-point, sampling 1 1
    __diffuse.r, 32-bit floating-point, sampling 1 1
```

### Versions

* OIIO branch/version: master
* OS: Cent 6
* C++ compiler: gcc 4.8.3

**Comment by lgritz** [comment]
Fixed by #1563

**Comment by kazuki-takahashi** [comment]
Thanks, Larry!
